### ISSUE TITLE
Add modifiers to interrupts code

### ISSUE THREAD

**Issue by lonestr9** [aevitas/BuddyWing.DefaultCombat#21]
There are times when i do and don't want the interrupts code to trigger, primarily during boss or epic enemy fights where they cannot be stunned / interrupted.  
Here is a sample of the modifiers I have added to the Assassin Hatred / Deception code:

//Interrupts
Spell.Cast("Jolt", ret => Me.CurrentTarget.IsCasting && !Me.CurrentTarget.HasDebuff("Unshakeable")),
Spell.Cast("Electrocute", ret => Me.CurrentTarget.IsCasting && !Me.CurrentTarget.HasDebuff("Boss Immunity")),
Spell.Cast("Low Slash", ret => Me.CurrentTarget.IsCasting && !Me.CurrentTarget.HasDebuff("Boss Immunity")),

I am sure others have modifiers they use as well, but this is a simple way to make sure you are only using these abilities where appropriate.  If others are interested in updating this, I will work on putting together a list for each spec if possible.

**Comment by alltrueist** [comment]
It would probably make the most sense to create a bool that checks for any of those buffs/debuffs (are they buffs or debuffs? Big difference).

Something like

``` csharp
private static bool canInterrupt
{
if Me.CurrentTarget.HasDebuff("Unshakeable")
return false;
if Me.CurrentTarget.HasDebuff("Boss Immunity")
return false;
return true;
}
```

You could then make the list as long as you want and even include specific spells you don't want to interrupt.

**Comment by lonestr9** [comment]
Yes alltrueist, sorry, these are buffs for the bosses.  Although I am not excited about a long list, I will begin the work as  I said I would :)  Might be a couple days to go through the specs, might be a list to start with out there somewhere.

**Comment by alltrueist** [comment]
Take your time! You've got the big ones right now. You might want to create two separate bools: canInterrupt and shouldInterrupt. In canInterrupt, you'd check to see if the boss is Unshakeable or has Boss Immunity (and that he's casting); in shouldInterrupt, you'd look for spells that we want to interrupt (heals, raid boss abilities, etc).
